boot/zephyr/nrf53_cpunet_ctl.c

```diff
@@ -6,32 +12,47 @@
 #include "bootutil/bootutil_log.h"
 MCUBOOT_LOG_MODULE_DECLARE(mcuboot);
 
-int do_network_core_update(void *addr, void *src_addr, size_t len)
+static void set_pcd_cmd_struct(void *src_addr, size_t len)
 {
-	struct pcd_cmd *rsp = (struct pcd_cmd *) PCD_RSP_ADDRESS;
-	if (addr == NULL) {
-		addr = (void *)PCD_CMD_ADDRESS;
-
-	}
-
-	struct pcd_cmd *cmd = (struct pcd_cmd *)addr;
-
-	rsp->magic = PCD_CMD_MAGIC_COPY;
+	struct pcd_cmd *cmd = (struct pcd_cmd *)PCD_CMD_ADDRESS;
 	cmd->magic = PCD_CMD_MAGIC_COPY;
 	cmd->src_addr = src_addr;
 	cmd->len = len;
 	cmd->offset = 0x10800;
-	nrf_spu_ramregion_set(NRF_SPU,
-			      APP_CORE_SRAM_SIZE/CONFIG_NRF_SPU_RAM_REGION_SIZE,
-			      true, NRF_SPU_MEM_PERM_READ, true);
+}
+
+static bool is_copying(void)
+{
+	struct pcd_cmd *cmd = (struct pcd_cmd *)PCD_CMD_ADDRESS;
+	if (cmd->magic == PCD_CMD_MAGIC_COPY) {
+		return true;
+	} 
+
+	return false;
+}
+
+static bool successful(void)
+{
+	struct pcd_cmd *cmd = (struct pcd_cmd *)PCD_CMD_ADDRESS;
+	if (cmd->magic != PCD_CMD_MAGIC_DONE) {
+		return false;
+	}
+	return true;
```

Comment:
Nit: It's weird that the parity of this function is opposite of `is_copying`. It could also be shorter.
```suggestion
	return (cmd->magic == PCD_CMD_MAGIC_DONE);
```

Reply:
True I'll fix it up it was sort of meant to be temporary.